internal/v2/remote_signer/remote_signer.go

```diff
@@ -72,15 +76,63 @@ func (s *remoteSigner) getStream() s2av2pb.S2AService_SetUpSessionClient {
 	return s.cstream
 }
 
-// getSignatureAlgorithm analyzes opts and determines signature algorithm to be
-// used.
-// TODO(rmehta19): fill in rest of logic(coverage of all signature algorithms).
-func getSignatureAlgorithm(opts crypto.SignerOpts) s2av2pb.SignatureAlgorithm {
-	if _, ok := opts.(*rsa.PSSOptions); ok {
-		return s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PSS_RSAE_SHA256
-	} else if opts == crypto.SHA256 {
-		return s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PKCS1_SHA256
-	} else {
-		return s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_UNSPECIFIED
+// getSignatureAlgorithm returns the signature algorithm that S2A must use when
+// performing a signing operation that has been offloaded by an application
+// using the crypto/tls libraries.
+func getSignatureAlgorithm(opts crypto.SignerOpts, leafCert *x509.Certificate) (s2av2pb.SignatureAlgorithm, error) {
```

Comment:
Please add the unit tests for this API in this PR. :)

Reply:
Will do!